common/src/main/java/org/apache/rocketmq/common/namesrv/NamesrvConfig.java

```diff
@@ -200,4 +221,68 @@ public boolean isEnableAllTopicList() {
     public void setEnableAllTopicList(boolean enableAllTopicList) {
         this.enableAllTopicList = enableAllTopicList;
     }
+
+    public boolean isStartupController() {
+        return isStartupController;
+    }
+
+    public void setStartupController(boolean startupController) {
+        isStartupController = startupController;
+    }
+
+    public int getControllerThreadPoolNums() {
+        return controllerThreadPoolNums;
+    }
+
+    public void setControllerThreadPoolNums(int controllerThreadPoolNums) {
+        this.controllerThreadPoolNums = controllerThreadPoolNums;
+    }
+
+    public int getControllerRequestThreadPoolQueueCapacity() {
+        return controllerRequestThreadPoolQueueCapacity;
+    }
+
+    public void setControllerRequestThreadPoolQueueCapacity(int controllerRequestThreadPoolQueueCapacity) {
+        this.controllerRequestThreadPoolQueueCapacity = controllerRequestThreadPoolQueueCapacity;
+    }
+
+    public String getControllerDLegerGroup() {
+        return controllerDLegerGroup;
+    }
+
+    public void setControllerDLegerGroup(String controllerDLegerGroup) {
+        this.controllerDLegerGroup = controllerDLegerGroup;
+    }
+
+    public String getControllerDLegerPeers() {
+        return controllerDLegerPeers;
+    }
+
+    public void setControllerDLegerPeers(String controllerDLegerPeers) {
+        this.controllerDLegerPeers = controllerDLegerPeers;
+    }
+
+    public String getControllerDLegerSelfId() {
+        return controllerDLegerSelfId;
+    }
+
+    public void setControllerDLegerSelfId(String controllerDLegerSelfId) {
+        this.controllerDLegerSelfId = controllerDLegerSelfId;
+    }
+
+    public String getControllerStorePath() {
+        return controllerStorePath;
+    }
+
+    public void setControllerStorePath(String controllerStorePath) {
+        this.controllerStorePath = controllerStorePath;
+    }
+
+    public boolean isEnableElectUncleanMaster() {
+        return enableElectUncleanMaster;
+    }
+
+    public void setEnableElectUncleanMaster(boolean enableElectUncleanMaster) {
+        this.enableElectUncleanMaster = enableElectUncleanMaster;
+    }
```

Comment:
Do we really need `setter` method? that will make code, which can get object reference, could change the state.

#### Describe in Chinese
该PR中的实体对象有必要定义 setter 方法吗？**setter方法意味着所有获取对象引用的代码都可以修改该对象状态**。而且似乎没有地方用到这些方法。

是否直接在构造函数中为 final 修饰的属性设置值更好、或者使用builder模式。

Reply:
@dugenkui03 
The setter method is required, otherwise the configuration will not take effect.
https://github.com/apache/rocketmq/blob/bd3059e76a00b0496ba0e2e433f6266f2ceb5683/namesrv/src/main/java/org/apache/rocketmq/namesrv/NamesrvStartup.java#L93